src/ixdat/techniques/spectroelectrochemistry.py

```diff
@@ -0,0 +1,250 @@
+        self.v (self.t), then first or last spectrum will be returned.
+
+        Args:
+            V (float): The potential at which to get the spectrum. Measurement.v must
+                be monotonically increasing for this to work.
+            t (float): The time at which to get the spectrum
+            index (int): The index of the spectrum
+            name (str): Optional. name to give the new spectrum if interpolated
+
+        Return Spectrum: The spectrum. The data is (spectrum.x, spectrum.y)
+        """
+        if V and V in self.v:  # woohoo, can skip interpolation!
+            index = int(np.argmax(self.v == V))
+        elif t and t in self.t:  # woohoo, can skip interpolation!
+            index = int(np.argmax(self.t == t))
+        if index:  # then we're done:
+            return self.spectrum_series[index]
+        # otherwise, we have to interpolate:
+        counts = self.spectra.data
+        end_spectra = (self.spectrum_series[0].y, self.spectrum_series[-1].y)
+        if V:
+            counts_interpolater = interp1d(
+                self.v, counts, axis=0, fill_value=end_spectra, bounds_error=False
+            )
+            # FIXME: This requires that potential and spectra have same tseries!
+            y = counts_interpolater(V)
+            name = name or f"{self.spectra.name}_{V}V"
+        elif t:
+            t_spec = self.spectra.axes_series[0].t
+            counts_interpolater = interp1d(
+                t_spec, counts, axis=0, fill_value=end_spectra, bounds_error=False
+            )
+            y = counts_interpolater(t)
+            name = name or f"{self.spectra.name}_{t}s"
+        else:
+            raise ValueError(f"Need t or V or index to select a spectrum!")
+
+        field = Field(
+            data=y,
+            name=name,
+            unit_name=self.spectra.unit_name,
+            axes_series=[self.wavelength],
+        )
+        return Spectrum.from_field(field, tstamp=self.tstamp)
+
+    def get_dOD_spectrum(
+        self,
+        V=None,
+        t=None,
+        index=None,
+        V_ref=None,
+        t_ref=None,
+        index_ref=None,
+    ):
+        """Return the delta optical density Spectrum given a point and reference point.
+
+        Provide exactly one of V, t, and index, and at most one of V_ref, t_ref, and
+        index_ref. For V and V_ref to work, the potential in the measurement must be
+        monotonically increasing.
+
+        Args:
+            V (float): The potential at which to get the spectrum.
+            t (float): The time at which to get the spectrum
+            index (int): The index of the spectrum
+            V_ref (float): The potential at which to get the reference spectrum
+            t_ref (float): The time at which to get the reference spectrum
+            index_ref (int): The index of the reference spectrum
+        Return Spectrum: The dOD spectrum. The data is (spectrum.x, spectrum.y)
```

Comment:
Returns:
    Spectrum: ...

Reply:
Oh, true. I've been consistently doing it wrong.